provider/pkg/internal/pulumiapi/deployment_settings.go

```diff
@@ -74,22 +76,87 @@ type GitHubConfiguration struct {
 	Paths               []string `json:"paths,omitempty"`
 }
 
-func (c *Client) CreateDeploymentSettings(ctx context.Context, stack StackName, ds DeploymentSettings) error {
+type SourceContext struct {
+	Git *SourceContextGit `json:"git,omitempty"`
+}
+
+type SourceContextGit struct {
+	RepoURL string         `json:"repoURL"`
+	Branch  string         `json:"branch"`
+	RepoDir string         `json:"repoDir,omitempty"`
+	Commit  string         `json:"commit,omitempty"`
+	GitAuth *GitAuthConfig `json:"gitAuth,omitempty"`
+}
+
+type GitAuthConfig struct {
+	PersonalAccessToken *SecretValue `json:"accessToken,omitempty"`
+	SSHAuth             *SSHAuth     `json:"sshAuth,omitempty"`
+	BasicAuth           *BasicAuth   `json:"basicAuth,omitempty"`
+}
+
+type SSHAuth struct {
+	SSHPrivateKey SecretValue  `json:"sshPrivateKey"`
+	Password      *SecretValue `json:"password,omitempty"`
+}
+
+type BasicAuth struct {
+	UserName SecretValue `json:"userName"`
+	Password SecretValue `json:"password"`
+}
+
+type SecretValue struct {
+	Value  string // Plaintext if Secret is false; ciphertext otherwise.
+	Secret bool
+}
+
```

Comment:
Doesn't this return both `Value` and `Ciphertext`?

Reply:
No, Pulumi Service only returns either plaintext or ciphertext, with the Secret bool indicating which one. But since we opted not to have the ability to return plaintext on these secret deployment settings, it only ever returns ciphertext.

This whole struct tree is taken from sdk, I unfrotunately had to copy it in to override the marshaling/unmarshaling logic